src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -185,11 +185,16 @@ def _process_node(self, base_node: NodeBase):
         if isinstance(base_node, SysPathChange):
             self._mutate_path_lookup(base_node)
         if isinstance(base_node, NotebookRunCall):
-            strpath = base_node.get_notebook_path()
-            if strpath is None:
-                yield DependencyProblem('dependency-not-constant', "Can't check dependency not provided as a constant")
+            paths = base_node.get_notebook_paths()
+            if None in paths:
+                yield DependencyProblem(
+                    'dependency-too-complex',
+                    f"Can't check dependency from {base_node.node.as_string()} because the expression is too complex",
```

Comment:
I would refrain from using "too complex" it is subjective, what makes something complex?

Reply:
changed